### PR TITLE
Add support up to Python 3.12

### PR DESCRIPTION
**Pull request by jcapiitao**
gh-91513: Added taskName attribute to logging module for use with asyncio tasks.
python/cpython#91513 [1]

credits to https://github.com/hrnciar

[1] https://github.com/python/cpython/issues/91513

### CODE REVIEW

**Comment by jcapiitao** [comment]
The patch works with Python 3.12 only.

**Comment by jcapiitao** [comment]
> The patch works with Python 3.12 only.

It's now backward compatible with older python releases

File: tox.ini [hunk 1]
```diff
@@ -37,4 +37,4 @@ ignore = E203,E501,W503,E231,A003,D101,D102,D103,D104,D105,D107,A003,D100
 
 [testenv:docs]
 deps = sphinx
-commands = python setup.py build_sphinx
\ No newline at end of file
+commands = sphinx-build -b html doc/source doc/build/html
```
Comment (jd) [inline tox.ini:40]: does not seems related?

**Review by jd — CHANGES_REQUESTED** [review]
You need to update Mergify configuration too.

File: tox.ini (hunk 1 above)
Comment (jcapiitao) [reply tox.ini:40]: Sphinx 7 has removed the setuptools integration https://github.com/sphinx-doc/sphinx/pull/11363
`setup.py` has been long deprecated and projects are encouraged to migrate away from using it.

Comment (jd) [reply tox.ini:40]: Ok, we should make that in a different PR though.

Comment (jcapiitao) [reply tox.ini]: PR proposed https://github.com/Mergifyio/daiquiri/pull/76

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -21,6 +21,7 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python }}
+          allow-prereleases: true
```
Comment (jcapiitao) [inline .github/workflows/ci.yaml:24]: I had to allow pre-releases in order to be able to fetch python 3.12 version, in setup-python action, which are not release as stable yet.

**Comment by jcapiitao** [comment]
Rebased on top of the main branch

**Review by sileht — DISMISSED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (sileht) [inline .github/workflows/ci.yaml:24]:
```suggestion
```

File: tox.ini [hunk 3]
```diff
@@ -10,7 +10,7 @@ commands =
   sh -c "rm errors.log everything.log"
 
 [testenv:pep8]
-basepython = python3.10
+basepython = python3.11
```
Comment (sileht) [inline tox.ini:13]:
```suggestion
basepython = python3.12
```

**Review by sileht — DISMISSED** [review]

**Review by sileht — APPROVED** [review]

**Pull request merged**